src/linker/Linker/MarkingHelpers.cs

```diff
@@ -11,12 +11,39 @@ public MarkingHelpers (LinkContext context)
 			_context = context;
 		}
 
-		public void MarkExportedType (ExportedType type, ModuleDefinition module, in DependencyInfo reason)
+		public void MarkMatchingExportedType (TypeDefinition typeToMatch, AssemblyDefinition assembly, in DependencyInfo reason)
 		{
-			if (!_context.Annotations.MarkProcessed (type, reason))
+			if (typeToMatch == null || assembly == null)
 				return;
-			if (_context.KeepTypeForwarderOnlyAssemblies)
-				_context.Annotations.Mark (module, new DependencyInfo (DependencyKind.ModuleOfExportedType, type));
+
+			ModuleDefinition module = assembly.MainModule;
+			if (module.GetMatchingExportedType (typeToMatch, out var exportedType)) {
+				MarkExportedType (exportedType, module, reason);
+				if (_context.Annotations.GetAction (assembly) == AssemblyAction.Copy) {
```

Comment:
Now that we're marking all exports and their scopes from copy assemblies, I don't think we need to check for copy here.

Reply:
I think this still matters in the case where you dynamically access a type forwarder in a link assembly, then the forwarder will be kept, but we could change the resolution scope. There's an example for this scenario in UsedTransitiveForwarderIsDynamicallyAccessed.cs.

Reply:
Clarified that I meant to remove the whole if statement. Thanks!

Reply:
Per offline conversation; whole block removed ;)